reconcilliation-test/cf_billing_reconcilliation.go

```diff
@@ -0,0 +1,197 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/alphagov/paas-billing/eventio"
+	cfclient "github.com/cloudfoundry-community/go-cfclient"
+)
+
+type listV3ServiceInstancesResponse struct {
+	Pagination cfclient.Pagination          `json:"pagination,omitempty"`
+	Resources  []cfclient.V3ServiceInstance `json:"resources,omitempty"`
+}
+
+type booleanSetOperationResults struct {
+	OnlyInFirstSet  map[string]bool
+	OnlyInSecondSet map[string]bool
+	SetIntersection map[string]bool
+}
+
+func get_boolean_results(firstSet map[string]bool, secondSet map[string]bool) booleanSetOperationResults {
+	var results booleanSetOperationResults
+	results.OnlyInFirstSet = make(map[string]bool)
+	results.OnlyInSecondSet = make(map[string]bool)
+	results.SetIntersection = make(map[string]bool)
+
+	for k, _ := range firstSet {
+		if secondSet[k] {
+			results.SetIntersection[k] = true
+		} else {
+			results.OnlyInFirstSet[k] = true
+		}
+	}
+	if len(results.SetIntersection) != len(secondSet) {
+		for k, _ := range secondSet {
+			if !firstSet[k] {
+				results.OnlyInSecondSet[k] = true
+			}
+		}
+	}
+	return results
+
+}
+func join_guids(guids map[string]bool) string {
+
+	var keys []string
+
+	for k, _ := range guids {
+		if k != "" {
+			keys = append(keys, k)
+		}
+	}
+	return (strings.Join(keys, ", "))
+}
+
+func main() {
+	//Do the billing api request
+	billingAPIURL, err := url.Parse(os.Getenv("BILLING_API_URL"))
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	billingAPIURL.Path = "/billable_events"
+
+	q := billingAPIURL.Query()
+	q.Set("range_start", time.Now().AddDate(0, 0, -2).Format("2006-01-02"))
```

Comment:
What's the relevance of the two-days-ago period start?

Reply:
So we get some data but don't go to the beginning of time. @schmie picked that date, so I can't speak to the actual date.